### PR TITLE
IGNITE-20077 Sql. Bump calcite version to 1.35.0

### PR DESCRIPTION
**Pull request by zstan**
Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/sql-engine/build.gradle [hunk 1]
```diff
@@ -57,6 +57,8 @@ dependencies {
     implementation libs.classgraph
     implementation libs.javassist
     implementation libs.auto.service.annotations
+    implementation libs.jsr305
```
Comment (AMashenkov) [inline modules/sql-engine/build.gradle:60]: Why do we need JSR305?

Comment (zstan) [reply modules/sql-engine/build.gradle:60]: for autogenerated code (@Nullable annotations) there is no way to change it into jetbrains implementation, previously this dependency was linked transiently

**Review by lowka — APPROVED** [review]

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/agg/AccumulatorsFactory.java [hunk 2]
```diff
@@ -284,9 +288,12 @@ public void add(RowT row) {
                 return;
             }
 
-            Object[] args = new Object[argList.size()];
-            for (int i = 0; i < argList.size(); i++) {
-                args[i] = handler.get(argList.get(i), row);
+            int params = literalAgg ? 1 : argList.size();
+
+            Object[] args = new Object[params];
```
[the remark this answers is not in the document]
Comment (zstan) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/agg/AccumulatorsFactory.java:293]: of course, fix it soon.

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/RexToLixTranslator.java [hunk 3]
```diff
@@ -301,12 +306,11 @@ Expression translateCast(
                 switch (sourceType.getSqlTypeName()) {
                     case CHAR:
                     case VARCHAR:
-                        convert = Expressions.call(BuiltInMethod.ST_GEOM_FROM_EWKT.method, operand);
-                        break;
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/RexToLixTranslator.java:305]: What reason for the change? we are supporting GEOMETRY?

**Review by ygerzhedovich — APPROVED** [review]

**Pull request merged**